homeassistant/components/netatmo/device_trigger.py

```diff
@@ -0,0 +1,155 @@
+"""Provides device automations for Netatmo."""
+from typing import List
+
+import voluptuous as vol
+
+from homeassistant.components.automation import AutomationActionType
+from homeassistant.components.device_automation import TRIGGER_BASE_SCHEMA
+from homeassistant.components.device_automation.exceptions import (
+    InvalidDeviceAutomationConfig,
+)
+from homeassistant.components.homeassistant.triggers import event as event_trigger
+from homeassistant.const import (
+    ATTR_DEVICE_ID,
+    CONF_DEVICE_ID,
+    CONF_DOMAIN,
+    CONF_ENTITY_ID,
+    CONF_PLATFORM,
+    CONF_TYPE,
+)
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant
+from homeassistant.helpers import config_validation as cv, entity_registry
+from homeassistant.helpers.typing import ConfigType
+
+from . import DOMAIN
+from .climate import STATE_NETATMO_AWAY, STATE_NETATMO_HG, STATE_NETATMO_SCHEDULE
+from .const import (
+    CLIMATE_TRIGGERS,
+    EVENT_TYPE_THERM_MODE,
+    INDOOR_CAMERA_TRIGGERS,
+    MODEL_NACAMERA,
+    MODEL_NATHERM1,
+    MODEL_NOC,
+    MODEL_NRV,
+    NETATMO_EVENT,
+    OUTDOOR_CAMERA_TRIGGERS,
+)
+
+CONF_SUBTYPE = "subtype"
+
+DEVICES = {
+    MODEL_NACAMERA: INDOOR_CAMERA_TRIGGERS,
+    MODEL_NOC: OUTDOOR_CAMERA_TRIGGERS,
+    MODEL_NATHERM1: CLIMATE_TRIGGERS,
+    MODEL_NRV: CLIMATE_TRIGGERS,
+}
+
+SUBTYPES = {
+    EVENT_TYPE_THERM_MODE: [
+        STATE_NETATMO_SCHEDULE,
+        STATE_NETATMO_HG,
+        STATE_NETATMO_AWAY,
+    ]
+}
+
+TRIGGER_TYPES = OUTDOOR_CAMERA_TRIGGERS + INDOOR_CAMERA_TRIGGERS + CLIMATE_TRIGGERS
+
+TRIGGER_SCHEMA = TRIGGER_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Required(CONF_TYPE): vol.In(TRIGGER_TYPES),
+        vol.Optional(CONF_SUBTYPE): str,
+    }
+)
+
+
+async def async_validate_trigger_config(hass, config):
+    """Validate config."""
+    config = TRIGGER_SCHEMA(config)
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(config[CONF_DEVICE_ID])
+
+    trigger = config[CONF_TYPE]
+
+    if (
+        not device
+        or device.model not in DEVICES
+        or trigger not in DEVICES[device.model]
+    ):
+        raise InvalidDeviceAutomationConfig
```

Comment:
Should we tell them what went wrong?

```suggestion
        raise InvalidDeviceAutomationConfig(f"Unsupported model {device.model}")
```

Reply:
That is always a good idea. Thanks.